### PR TITLE
Improve `Random.getstate` and `Random.setstate`

### PR DESCRIPTION
**Pull request by sobolevn**
Source: https://github.com/python/cpython/blob/243436f3779c1e7bed1fd4b23d5a8ec5eff40699/Lib/random.py#L168

Refs https://github.com/python/typeshed/blob/4b75ecbbe2baf600847b79c3e4c8ffd97ff6cae5/stdlib/_random.pyi#L2

### CODE REVIEW

**Review by hauntsaninja — COMMENTED** [review]
Hmm, this feels potentially a little bit fragile, since the return type is versioned. That said, for the state versions that are currently supprted (2 & 3), the return type is about the same (the only difference is version 2 uses signed ints and version 3 uses unsigned). Ideally I think we'd have a `Tuple[int, Any, ...]` type, but that's not legal.

Also linking #6063, for some previous discussion of these methods, where the argument is made that these should be opaque types.

All in all I'm in favour of this, but figured I'd mention in case you see something worth changing or some clever solution.

**Comment by sobolevn** [comment]
I thought about `VERSION` as `Literal[2]` and `Literal[3]`, but it feels too complex with almost no benefits.

I also think that concrete `tuple` is better than just `tuple[Any, ...]`. If someone wants to redefine this methos, `# type: ignore` is there to help.

File: stdlib/random.pyi [hunk 1]
```diff
@@ -9,15 +9,16 @@ _T = TypeVar("_T")
 
 class Random(_random.Random):
     VERSION: ClassVar[int]
+    gauss_next: float | None  # undocumented
```
Comment (JelleZijlstra) [inline stdlib/random.pyi:12]: Do you have a use for this? We have a policy to add undocumented things if users ask for them, but we shouldn't go out of our way to add every undocumented attribute.

**Comment by JelleZijlstra** [comment]
I feel like this might be better as just `object`. The exact return type is clearly an implementation detail, and I don't see much point in describing the precise format in typeshed.

**Comment by srittau** [comment]
Using `NewType` is also an option.

File: stdlib/random.pyi (hunk 1 above)
Comment (sobolevn) [reply stdlib/random.pyi:12]: Yes, it is quite widely used: https://cs.github.com/?q=.gauss_next%20language%3APython&scopeName=All%20repos

I personally remember using in it my fake-data generator project.

**Comment by sobolevn** [comment]
> The exact return type is clearly an implementation detail, and I don't see much point in describing the precise format in typeshed.

Yes, I agree. We will just generate `overload` errors for users with alterantive implementations.

**Pull request closed without merging**